acceptance/tests/connect/connect_proxy_lifecycle_test.go

```diff
@@ -0,0 +1,218 @@
+		{secure: false, helmValues: map[string]string{}},
+		{secure: true, helmValues: map[string]string{}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "true",
+			helmGracePeriodSecondsKey: "15",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "true",
+			helmGracePeriodSecondsKey: "15",
+		}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "15",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "15",
+		}},
+		{secure: false, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "0",
+		}},
+		{secure: true, helmValues: map[string]string{
+			helmDrainListenersKey:     "false",
+			helmGracePeriodSecondsKey: "0",
+		}},
+	} {
+		// Determine if listeners should be expected to drain inbound connections
+		var drainListenersEnabled bool
+		val, ok := testCfg.helmValues[helmDrainListenersKey]
+		if ok {
+			drainListenersEnabled, err = strconv.ParseBool(val)
+			require.NoError(t, err)
+		}
+
+		// Determine expected shutdown grace period
+		var gracePeriodSeconds int64
+		val, ok = testCfg.helmValues[helmGracePeriodSecondsKey]
+		if ok {
+			gracePeriodSeconds, err = strconv.ParseInt(val, 10, 64)
+			require.NoError(t, err)
+		} else {
+			// Half of the helm default to speed tests up
+			gracePeriodSeconds = 15
+		}
+
+		name := fmt.Sprintf("secure: %t, drainListeners: %t, gracePeriodSeconds: %d", testCfg.secure, drainListenersEnabled, gracePeriodSeconds)
+		t.Run(name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+			releaseName := helpers.RandomName()
+
+			connHelper := connhelper.ConnectHelper{
+				ClusterKind: consul.Helm,
+				Secure:      testCfg.secure,
+				ReleaseName: releaseName,
+				Ctx:         ctx,
+				Cfg:         cfg,
+				HelmValues:  testCfg.helmValues,
+			}
+
+			connHelper.Setup(t)
+			connHelper.Install(t)
+			connHelper.DeployClientAndServer(t)
+
+			// TODO: should this move into connhelper.DeployClientAndServer?
+			logger.Log(t, "waiting for static-client and static-server to be registered with Consul")
+			retry.Run(t, func(r *retry.R) {
+				for _, name := range []string{
+					"static-client",
+					"static-client-sidecar-proxy",
+					"static-server",
+					"static-server-sidecar-proxy",
+				} {
+					logger.Logf(t, "checking for %s service in Consul catalog", name)
+					instances, _, err := connHelper.ConsulClient.Catalog().Service(name, "", nil)
+					r.Check(err)
+
+					if len(instances) != 1 {
+						r.Errorf("expected 1 instance of %s", name)
+					}
+				}
+			})
+
+			if testCfg.secure {
+				connHelper.TestConnectionFailureWithoutIntention(t)
+				connHelper.CreateIntention(t)
+			}
+
+			connHelper.TestConnectionSuccess(t)
+
+			// Get static-client pod name
+			// TODO: is this necessary?
```

Comment:
I think this todo can be removed? We do seem to use the client pod name below.

Reply:
Yes, this one too